docs/PLUGINS.md

```diff
@@ -311,3 +311,32 @@ public class ConfigurationAsCodeTest {
     }
 }
 ```
+
+### JSON Schema Test
+We generate a JSON schema that users can use to validate their changes and provide IDE assistance, 
+you can test that your plugin's example yaml file validates correctly by implementing the below test:
+`SchemaGenerationTest` provides a abstraction layer to test out the plugins YAML file against the generated schema.
+
+You can test if your YAML file validates against the schema.
+
+Step 1
+
+Create a YAML file for the configurators corresponding to the developed plugin.
+For eg: `validJenkinsConfigurator.yml`
+```yaml
+jenkins:
+    systemMessage: "Configured by Configuration as Code plugin"
+    numExecutors: "Hello"
+
+```
+
+Step 2
+
+Add a test for the YAML file
+
+```java
+@Test
```

Comment:
formatting is off:
![image](https://user-images.githubusercontent.com/21194782/66395416-4b292300-e9cf-11e9-91b2-021bf48a12de.png)